fullnode-messages/src/index.ts

```
@@ -0,0 +1,194 @@
+    });
+  }
+
+  private onOpen(): void {
+    this.subscribeMessages()
+    this.heartbeat = setInterval(() => {
+      this.sendPing();
+    }, this.heartbeatInterval);
+  }
+
+  private onClose(): void {
+    if (this.ws) {
+      this.ws.close();
+      this.ws = null;
+    }
+
+    setTimeout(() => {
+      this.start();
+    }, this.retryConnectionInterval);
+
+    if (this.heartbeat) {
+      clearInterval(this.heartbeat);
+      this.heartbeat = null;
+      this.latestPingDate = null;
+    }
+  }
+
+  private sendPing(): void {
+    if (this.latestPingDate) {
+      // Skipping sendPing. Still waiting for pong...
+      return;
+    }
+
+    const msg = JSON.stringify({'type': 'ping'})
+    this.latestPingDate = new Date();
+    this.timeoutTimer = setTimeout(() => this.onClose(), this.connectionTimeout);
+    this.sendMessage(msg);
+  }
+
+  private onPong(): void {
+    this.latestPingDate = null;
+
+    if (this.timeoutTimer) {
+      clearTimeout(this.timeoutTimer);
+      this.timeoutTimer = null;
+    }
+  }
+
+  private setState(state: number): void {
+    this.state = state;
+    this.emit('state', state);
+    console.log('State updated to', state);
+  }
+
+  private handleMessage(data: wsDataType): void {
+    console.log('Message', data.type)
+    switch(data.type) {
+      case 'subscribed':
+        this.onSubscribed(data);
+        break;
+      case 'pong':
+        this.onPong();
+        break;
+      case 'network:new_tx_accepted':
+        this.onNewTx(data);
+        break;
+      default:
+        break;
+    }
+  }
+
+  private onNewTx(data: wsDataType): void {
+    const deduplicationId = `new-tx-${data.tx_id as string}`;
+    const params = {
+      MessageBody: JSON.stringify(data),
+      QueueUrl: QUEUE_URL,
+      MessageGroupId: 'wallet-service-new-tx',
+      MessageDeduplicationId: deduplicationId,
+    };
+
+    sqs.sendMessage(params, function(err, data) {
+      if (err) console.log(err, err.stack); // an error occurred
```

Maybe we should put the data in an internal queue and retry in case of error. If the queue is full, we can raise an unrecoverable error (or start syncing again after the SQS is back).